DataStructures.Tests/SegmentTrees/SegmentTreeUpdateTest.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+using DataStructures.SegmentTrees;
+using NUnit.Framework;
+
+namespace DataStructures.Tests.SegmentTrees
+{
+    [TestFixture]
+    public class SegmentTreeUpdateTests
+    {
+        private readonly SegmentTreeUpdate testTree = new SegmentTreeUpdate(new [] {8, 9, 1, 4, 8, 7, 2});
+
+        [TestCase(2, 3, 1, 4, 24)]
+        [TestCase(0, 3, 1, 4, 24)]
```

Comment:
It's really not obvious at all that we get 24 in the second test case because of the first test case. Please update tests so that they don't influence each other.